core/src/impl/Kokkos_Half_MathematicalFunctions.hpp

```
@@ -315,6 +321,100 @@ KOKKOS_INLINE_FUNCTION bool isnan(Kokkos::Experimental::bhalf_t x) {
+
+   // Handle zeros
+   if (from == fp16_t(0)) {
+     // from is +0.0 or -0.0
+     // Return smallest magnitude number with the sign of 'to'.
+     // nextafter(±0, negative) -> smallest_negative
+     // nextafter(±0, positive) -> smallest_positive
+     return bit_cast<fp16_t>((to > from) ? FP16_SMALLEST_POS_DN
+                                         : FP16_SMALLEST_NEG_DN);
+   }
+
+   // Determine direction and sign of 'from'
+   // True if moving to positive infinity
+   bool to_positive_infinity = (to > from);
+   bool from_is_negative     = (uint_from & FP16_SIGN_MASK);
+
+   std::uint16_t uint_result = uint_from + 2 * (to_positive_infinity ^ from_is_negative) - 1;
+   // This is equivalent to the following operations.
+   // std::uint16_t uint_result;
+   //
+   // if (from_is_negative) {
+   //   // For negative numbers, increasing magnitude means moving towards -inf
+   //   // (larger uint value) Decreasing magnitude means moving towards zero
+   //   // (smaller uint value)
+   //   if (to_positive_infinity) {
+   //     // Moving toward zero or positive
+   //     uint_result = uint_from - 1;
+   //   } else {
+   //     // Moving toward negative infinity
+   //     uint_result = uint_from + 1;
+   //   }
+   // } else {
+   //   // For positive numbers, increasing magnitude means moving towards +inf
+   //   // (larger uint value) Decreasing magnitude means moving towards zero
+   //   // (smaller uint value)
+   //   if (to_positive_infinity) {
+   //     // Moving toward positive infinity
+   //     uint_result = uint_from + 1;
+   //   } else {
+   //     // Moving toward zero or negative infinity
+   //     uint_result = uint_from - 1;
+   //   }
+   // }
+   fp16_t result = bit_cast<fp16_t>(uint_result);
+   KOKKOS_IF_ON_HOST((
+       if (isinf(result)) {
+         // if from is finite, but the expected result is an infinity, raises FE_INEXACT and FE_OVERFLOW.
+         std::feraiseexcept(FE_INEXACT | FE_OVERFLOW);
```

I am not so keen on doing C-style floating-point exception handling.
If we did decide to get into it, then we would have to test for it.

---

If you are fine, I would like to suppress this exception handling. 
Since we can only raise exception on Host, it may be not that useful. 
In addition, this makes the behavior of this function on Host and Device in-consistent.
Can I just leave comment and remove the exception?

@dalg24 @cedricchevalier19